main_penalty_calculate/builders/violators_avaliations_builder.py

```diff
@@ -1,77 +1,107 @@
+from datetime import datetime
+
 from ..models import ViolatorAvaliation, IdentityCard
 
 
 class ViolatorsAvaliationsBuilder:
     _LICENSE_PLATE_IN_LIST = 0
     _FIRST_TRAFFIC_VIOLATION = 0
+    _DATE_FORMAT = "%Y-%m-%d %H:%M:%S"
+    _VALIDITY_PERIOD_OF_INFRINGEMENT = 30
+    _INVALID_DEMERIT_POINTS = 0
+    _INFRACTION_PENALTIES = {
+        'Leve': 3,
+        'Média': 4,
+        'Grave': 5,
+        'Gravíssima': 7
+    }
 
     def __init__(self, traffic_violations):
         self._traffic_violations = traffic_violations
+        self._violators_avaliations = []
 
-    def _is_identity_card_number_already_present(
-        self,
-        identity_card_number_to_be_checked,
-        identity_card_number_list
-    ):
-        for identity_card in identity_card_number_list:
+    def _is_identity_card_number_already_present(self):
+        for self._violator in self._violators_avaliations:
             if (
-                identity_card.identity_card_number ==
-                identity_card_number_to_be_checked
+                self._violator.identity_card_number ==
+                self._traffic_violation.identity_card_number
             ):
                 return True
         return False
 
     def _is_license_plate_number_already_present(
         self,
-        license_plate_number_to_be_checked,
         license_plate_list
     ):
         for license_plate_number in license_plate_list:
             if (
                 license_plate_number ==
-                license_plate_number_to_be_checked
+                self._traffic_violation.license_plate_number
             ):
                 return True
         return False
 
-    def _agroup_license_plates(
-        self,
-        license_plate_in_review,
-        violators_list,
-        violator_in_review
-    ):
-        for violator in violators_list:
-            if violator.identity_card_number == violator_in_review:
-                if not self._is_license_plate_number_already_present(
-                    license_plate_in_review,
-                    violator.license_plate_numbers
-                ):
-                    violator.license_plate_numbers.append(
-                        license_plate_in_review
-                    )
+    def _aggregate_license_plates(self):
+        if not self._is_license_plate_number_already_present(
+            self._violator.license_plate_numbers
+        ):
+            self._violator.license_plate_numbers.append(
+                self._traffic_violation.license_plate_number
+            )
+
+    def _aggregate_demerit_points(self):
+        self._violator.sum_demerit_points(
+            self._INFRACTION_PENALTIES[
+                self._traffic_violation.type_infraction
+            ]
+        )
+
+    def _is_demerit_points_valid(self):
+        if (
```

Comment:
`if condicao return True` nao faz muito sentido nao eh mesmo?
Lembra como `boolean` funciona

Reply:
Não entendi pq não faz sentido. A função diz se um valor é menor ou igual a outro, se sim retorna True e se não for retorna False, assim como a do RG retorna True se um valor for igual a outro e False caso não seja.

Reply:
Faz a execucao do Python passo a passo e vc vai ver que existe uma redundancia no seu algoritmo

Reply:
Certo! Eu encontrei algumas redundâncias no código, como está agora?

Reply:
Vc nao mudou esse metodo, ele ainda apresenta redundancia.

Reply:
Vc nao mudou esse metodo, ele ainda apresenta redundancia.

Reply:
Hmm... Então o `<=` seria o problema? Ele é uma outra condição junto ao `if`?

Reply:
Boa!